File: apple/Sources/ReactNativeWatchOS/Modifiers/LabeledContentStyle.swift
import SwiftUI

/// SwiftUI `.labeledContentStyle(_:)`. `LabeledContent` and its
/// `.automatic` style are available on watchOS 9 (the deployment target),
/// so no availability gating is required. Only `.automatic` is exposed —
/// any other value also resolves to `.automatic`.
enum RNWLabeledContentStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("labeledContentStyle") { view, _, _ in
            AnyView(view.labeledContentStyle(.automatic))
        }
    }
}

File: src/modifiers/labeledContentStyle.ts
import { createModifier } from './createModifier';

export type LabeledContentStyle = 'automatic';

/// SwiftUI `.labeledContentStyle(_:)`. Sets the visual style applied to
/// `LabeledContent` within the view (watchOS 9+). `'automatic'` defers to
/// the platform default.
export function labeledContentStyle(style: LabeledContentStyle = 'automatic') {
  return createModifier('labeledContentStyle', { style });
}
